src/sema/types.rs

```diff
@@ -845,6 +850,17 @@ impl Type {
         }
     }
 
+    pub fn is_rational(&self) -> bool {
+        match self {
+            Type::Rational => true,
+            Type::Ref(r) => r.is_rational(),
+            Type::StorageRef(_, r) => r.is_rational(),
+            Type::Int(_) => true,
+            Type::Uint(_) => true,
+            Type::Value => true,
```

Comment:
Int/Uint/Value is not rational.

Reply:
I get error message if I remove Int/Uint/Value.

```
test.sol:4:26-28: error: expected ‘uint256’, found rational
Line 4:
	uint x = .5 * 8;
	---------^^
thread 'solana_tests::simple::returns' panicked at 'called `Option::unwrap()` on a `None` value', src/emit/mod.rs:4348:54
```
There may be something wrong with my cast then.